lib/cc/analyzer/engine.rb

```diff
@@ -16,7 +16,9 @@ def initialize(name, metadata, code_path, config, label)
         @label = label.to_s
       end
 
-      def run(stdout_io, stderr_io = StringIO.new)
+      def run(stdout_io, stderr_io = StringIO.new, container_log = NullContainerLog.new)
```

Comment:
What do you think of making these named params?

```ruby
def run(stdout_io:, stderr_io: StringIO.new, container_log: NullContainerlog.new
```

Then you could call it below with just

```ruby
run(stdout_io: @formatter, container_log: container_log)
```

and not have to worry about the stderr value